Simbody/src/CableSpan.cpp

```diff
@@ -0,0 +1,3028 @@
+            dataPos.X_GS = calcSurfaceFrameInGround(state);
+            // Store the geodesic's frenet frames in ground frame.
+            dataPos.X_GP = dataPos.X_GS.compose(dataInst.X_SP);
+            dataPos.X_GQ = dataPos.X_GS.compose(dataInst.X_SQ);
+        }
+
+        getSubsystem().markCacheValueRealized(state, m_indexDataPos);
+
+        return dataPos;
+    }
+
+    //------------------------------------------------------------------------------
+    //  Generating Geodesic Points For Visualization
+    //------------------------------------------------------------------------------
+
+    void calcDecorativePathPoints(
+        const State& state,
+        const std::function<void(Vec3 point_G)>& sink) const
+    {
+        if (!isInContactWithSurface(state)) {
+            return;
+        }
+
+        const CurveSegmentData::Instance& dataInst = getDataInst(state);
+        const Transform& X_GS                      = getDataPos(state).X_GS;
+
+        // Lambda for computing the point in ground from a GeodesicKnotPoint,
+        // and writing it to the sink.
+        auto calcPathPointAndWriteToSink =
+            [&](const ContactGeometry::GeodesicKnotPoint& q)
+        {
+            const Vec3 point_G = X_GS.shiftFrameStationToBase(q.point);
+            sink(point_G);
+        };
+
+        if (getContactGeometry().isAnalyticFormAvailable()) {
+            // TODO Depending on what is actually desired by the end-user, this
+            // might need revision.
+
+            // For analytic surfaces there is no numerical integrator to give a
+            // natural interspacing of knots. So instead I just assume you
+            // would like some OK granularity. This can be done by interspacing
+            // the knots by a certain "angle" (e.g. 5 degrees is pretty fine),
+            // and using the radius of curvature to convert the angle to a
+            // spatial interspacing of the samples.
+            constexpr Real c_AngularSpacingInDegrees = 5.;
+
+            // To convert the angular spacing to a spatial spacing we need the
+            // curvature. The curvature is generally not constant along a
+            // geodesic, but there are not a lot of analytic surfaces at the
+            // moment (sphere and cylinder?). For these we can just look at the
+            // curvature at the boundary frames to know the curvature along the
+            // entire geodesic.
+            const Real maxCurvature_P = max(abs(dataInst.curvatures_P));
+            const Real maxCurvature_Q = max(abs(dataInst.curvatures_P));
+            const Real minRadiusOfCurvature =
+                1. / std::max(maxCurvature_P, maxCurvature_Q);
+
+            // Estimate the spatial interspacing from the angular spacing and
+            // the radius of curvature.
+            const Real lengthInterspacing =
+                minRadiusOfCurvature * c_AngularSpacingInDegrees / 180. * Pi;
```

Comment:
SimTK_DEG2RAD or similar

Reply:
I wrote the `c_AngularInterspacing = 0.087` in radians instead, commenting that it equals 5 degrees